Classes/Controller/LoginController.php

```diff
@@ -211,13 +211,15 @@ protected function getAuth0(): Auth0
         if ($this->auth0 instanceof Auth0) {
             return $this->auth0;
         }
-
         if ($this->extensionConfiguration->isGenericCallback()) {
             $callback = $this->getCallback('login');
         } else {
             trigger_error('Using callback settings for frontend request is deprecated as there is a dedicated callback middleware.', E_USER_DEPRECATED);
             $uri = $GLOBALS['TYPO3_REQUEST']->getUri();
             $referrer = sprintf('%s://%s%s', $uri->getScheme(), $uri->getHost(), $uri->getPath());
+            if ($this->settings['referrerAnchor']) {
+                $referrer .= '#' . $this->settings['referrerAnchor'];
```

Comment:
Maybe add a security check here for only allowed characters.

Reply:
Since the value will be set by editors, i don't see the need for a special check here. Maybe some extra validation via TCA.